client/server/middleware/logger.js

```diff
@@ -1,43 +1,54 @@
+/**
+ * External dependencies
+ */
+import uaParser from 'ua-parser-js';
+
 /**
  * Internal dependencies
  */
 import { getLogger } from 'server/lib/logger';
+import config from 'config';
 
 const NS_TO_MS = 1e-6;
 
+const parseUA = ( rawUA ) => {
+	const parsedUA = uaParser( rawUA );
+	if ( parsedUA.browser.name && parsedUA.browser.version ) {
+		const version = parsedUA.browser.version.split( '.' )[ 0 ];
+		return `${ parsedUA.browser.name } ${ version }`;
+	}
+	return rawUA;
+};
+
 const logRequest = ( req, res, options ) => {
-	const { requestStart } = options;
+	const { requestStart, env, version } = options;
 
-	let fields = {
+	const fields = {
 		method: req.method,
 		status: res.statusCode,
 		length: res.get( 'content-length' ),
 		url: req.originalUrl,
 		duration: Number(
 			( Number( process.hrtime.bigint() - requestStart ) * NS_TO_MS ).toFixed( 3 )
 		),
+		httpVersion: req.httpVersion,
+		version,
+		env,
+		userAgent: parseUA( req.get( 'user-agent' ) ),
 	};
 
-	if ( process.env.NODE_ENV === 'production' ) {
-		// Standard Apache combined log output minus the remote-user
-		fields = {
-			...fields,
-			remoteAddr: req.ip,
-			httpVersion: req.httpVersionMajor + '.' + req.httpVersionMinor,
-			referrer: req.get( 'referer' ),
```

Comment:
Would be cool if you can provide some background on why we're dropping `referrer` and `remoteAddr`.

Reply:
Not a strong reason to be honest. 

I have never actually used those fields to diagnose any performance or functional problem, and logging them could be a bit intrusive, although our [Privacy Policy](https://automattic.com/privacy/) explicitly mention both being logged.

I can bring them back if you think they can be useful.

Reply:
I recall I've used `referrer` for auditing something before. I'd suggest keeping it.

Not sure what the `remoteAddr` value could be, but perhaps it could be useful if we're auditing with regards to a possible DoS/DDoS.

Maybe let's keep both for now.